### PR TITLE
Fce 817 rtp metrics

### PR DESCRIPTION
**Pull request by roznawsk**

### CODE REVIEW

File: ex_webrtc/integration_test/test_videoroom/lib/test_videoroom/application.ex [hunk 1]
```diff
@@ -7,7 +7,13 @@ defmodule TestVideoroom.Application do
...
     children = [
       {Phoenix.PubSub, name: TestVideoroom.PubSub},
-      TestVideoroomWeb.Endpoint
+      TestVideoroomWeb.Endpoint,
+      {Membrane.TelemetryMetrics.Reporter,
+       [
+         metrics: Membrane.RTC.Engine.Endpoint.ExWebRTC.Metrics.metrics(),
+         name: ExWebrtcMetricsReporter
+       ]},
+      TestVideoroom.MetricsScraper
```
Comment (Karolk99) [inline ex_webrtc/integration_test/test_videoroom/lib/test_videoroom/application.ex:16]: What is the advantage of using Membrane.TelemetryMetrics instead of regular telemetry in case of ex_webrtc_endpoint ?

File: ex_webrtc/lib/ex_webrtc/peer_connection_handler/metrics.ex [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        :pli_count,
+        :markers_sent,
+        :codec,
+        :retransmitted_packets_sent,
+        :retransmitted_bytes_sent
+      ]),
+      %{},
+      telemetry_label
+    )
+  end
```
Comment (Karolk99) [inline ex_webrtc/lib/ex_webrtc/peer_connection_handler/metrics.ex:250]: How does it work in the case of simulcast?

File: ex_webrtc/lib/ex_webrtc/peer_connection_handler.ex [hunk 3]
```diff
@@ -26,6 +27,16 @@ defmodule Membrane.RTC.Engine.Endpoint.ExWebRTC.PeerConnectionHandler do
                 spec: [PeerConnection.Configuration.ice_server()],
                 description: "List of servers that may be used by ICE agent",
                 default: []
+              ],
+              telemetry_label: [
+                spec: Membrane.TelemetryMetrics.label(),
+                default: [],
+                description: "Label passed to Membrane.TelemetryMetrics functions"
+              ],
+              get_stats_interval: [
```
Comment (Karolk99) [inline ex_webrtc/lib/ex_webrtc/peer_connection_handler.ex:36]: What about config.exs instead of endpoint option

**Review by Karolk99 — CHANGES_REQUESTED** [review]
Also integration tests are failing

File: ex_webrtc/lib/ex_webrtc/peer_connection_handler/metrics.ex (hunk 2 above)
Comment (roznawsk) [reply ex_webrtc/lib/ex_webrtc/peer_connection_handler/metrics.ex]: We track each variant separately (the `track_identifier` has a variant in its name)

File: ex_webrtc/integration_test/test_videoroom/lib/test_videoroom/application.ex (hunk 1 above)
Comment (roznawsk) [reply ex_webrtc/integration_test/test_videoroom/lib/test_videoroom/application.ex:16]: The advantage of using Membrane.TelemetryMetrics over regular Telemetry.Metrics, especially in the context of ex_webrtc_endpoint, stems from the specific adaptations in Membrane.TelemetryMetrics that are tailored for multimedia applications. Felix mentioned that regular Telemetry.Metrics might not be sufficiently suited for multimedia use cases. Furthermore, Membrane.TelemetryMetrics includes certain optimizations or "hacks" that you would need to manually implement if you were using Telemetry.Metrics, enhancing efficiency and effectiveness for multimedia-specific metrics collection and monitoring in the Membrane framework.

File: ex_webrtc/integration_test/test_videoroom/config/config.exs [hunk 4]
```diff
@@ -33,16 +33,15 @@ config :logger, :console,
   format: "$time $metadata[$level] $message\n",
   metadata: [:request_id]
 
-config :logger,
-  compile_time_purge_matching: [
-    [level_lower_than: :info]
-  ]
```
Comment (Karolk99) [inline ex_webrtc/integration_test/test_videoroom/config/config.exs:39]: Why did you remove it?

File: ex_webrtc/lib/ex_webrtc/peer_connection_handler.ex [hunk 5]
```diff
@@ -290,8 +321,14 @@ defmodule Membrane.RTC.Engine.Endpoint.ExWebRTC.PeerConnectionHandler do
...
-  defp handle_webrtc_msg({:signaling_state_change, :stable}, _ctx, state) do
-    {[notify_parent: :negotiation_done], state}
+  defp handle_webrtc_msg({:signaling_state_change, new_state}, _ctx, state) do
+    actions =
+      case {state.peer_connection_signaling_state, new_state} do
+        {:have_remote_offer, :stable} -> [notify_parent: :negotiation_done]
+        _other -> []
+      end
+
+    {actions, %{state | peer_connection_signaling_state: new_state}}
```
Comment (Karolk99) [inline ex_webrtc/lib/ex_webrtc/peer_connection_handler.ex:331]: Could you tell me something more about this part?

File: ex_webrtc/integration_test/test_videoroom/config/config.exs (hunk 4 above)
Comment (roznawsk) [reply ex_webrtc/integration_test/test_videoroom/config/config.exs:39]: It's quite often that for tests I have to add debug logs. So that means both removing this entry (`compile_time_purge_matching`) and `logger: level`.

After this change there is a single place to change log level (`logger: level`)

File: ex_webrtc/lib/ex_webrtc/peer_connection_handler.ex (hunk 5 above)
Comment (roznawsk) [reply ex_webrtc/lib/ex_webrtc/peer_connection_handler.ex:331]: The change of state to: `{:signaling_state_change, :stable}` means, that the connection is in a stable state (i.e. there are no ongoing negotiations, tracks added/ removed etc.)

This usually means, that the ongoing negotiation has been completed.
However, there is a single instance when this event is received, even though there has been no negotiation going on. That happens at the very beginning. Upon creation the PeerConnection has state `:new`, which then changes to `:stable` (even if no tracks are negotiated). So far we ignored this case, which was a bug. Now it's fixed, by checking if there has been a negotiation going on before receiving the `:stable` state. 

For more info check MDN docs: https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/signalingState

File: ex_webrtc/integration_test/test_videoroom/config/config.exs (hunk 4 above)
Comment (Karolk99) [reply ex_webrtc/integration_test/test_videoroom/config/config.exs:39]: I would still leave it as it is because the engine and endpoints create a lot of debug logs, and this config removes them at compile time instead of ignoring invoked calls. It's about performance.

**Review by Karolk99 — APPROVED** [review]

File: ex_webrtc/integration_test/test_videoroom/config/config.exs (hunk 4 above)
Comment (roznawsk) [reply ex_webrtc/integration_test/test_videoroom/config/config.exs:39]: It's only for tests, but I can revert this change.
